lib/web_ui/lib/src/engine/canvaskit/image.dart

```diff
@@ -43,8 +43,14 @@ class CkAnimatedImage implements ui.Image {
   // being garbage-collected, or by an explicit call to [delete].
   late final SkiaObjectBox box;
 
-  CkAnimatedImage(this._skAnimatedImage) {
-    box = SkiaObjectBox(this, _skAnimatedImage as SkDeletable);
+  CkAnimatedImage(SkAnimatedImage skAnimatedImage) : this._(skAnimatedImage, null);
+
+  CkAnimatedImage._(this._skAnimatedImage, SkiaObjectBox? boxToClone) {
+    if (boxToClone != null) {
+      box = boxToClone.clone(this);
```

Comment:
Should we assert that the image in `boxToClone` and `_skAnimatedImage` are the same?

Alternatively, we could split this into two specialized constructors.